pandas/_libs/src/vendored/numpy/datetime/np_datetime.c

```diff
@@ -430,6 +433,15 @@ npy_datetime npy_datetimestruct_to_datetime(NPY_DATETIMEUNIT base,
   }
 
   const int64_t days = get_datetimestruct_days(dts);
+  if (days == -1) {
+    PyGILState_STATE gstate = PyGILState_Ensure();
+    bool did_error = PyErr_Occurred() == NULL ? false : true;
+    PyGILState_Release(gstate);
+    if (did_error) {
+      return -1;
+    }
+  }
```

Comment:
this is acquiring the gil, checking if an overflow happened, then releasing it?  is there a nogil alternative?

id be more comfortable with this if we had a test where it made a difference

Reply:
The GIL is only acquired when `days == -1` because you need to check if that is a legitimate value or if a Python exception was thrown. This is essentially what `except? -1` does in Cython